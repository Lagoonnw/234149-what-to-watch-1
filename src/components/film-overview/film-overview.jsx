import React, {Fragment}      from 'react';
import {convertReviewRaiting} from "../../helpers/convert-review-raiting/convert-review-raiting";

export const FilmOverview = ({description, rating, director, starring, scoresCount}) => {
  return (
    <Fragment>
      <div className="movie-rating">
        <div className="movie-rating__score">{`${rating.toFixed(1)}`}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{`${convertReviewRaiting(rating)}`}</span>
          <span className="movie-rating__count">{`${scoresCount} ratings`}</span>
        </p>
      </div>

      <div className="movie-card__text">
        <p>{description}</p>

        <p className="movie-card__director"><strong>{`Director: ${director}`}</strong></p>

        <p className="movie-card__starring"><strong>{`Starring: ${starring.join(`, `)}`}</strong></p>
      </div>
    </Fragment>
  );
};

