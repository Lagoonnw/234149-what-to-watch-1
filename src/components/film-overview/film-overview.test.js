import React from 'react';
import renderer from 'react-test-renderer';
import {FilmOverview} from './film-overview';

const mockProps = {
  description: `Some text`,
  rating: 8.9,
  director: `John Dow`,
  starring: [`Actor`, `Actress`],
  scoresCount: 894
};

test(`Should render Film overview correctly`, () => {
  const tree = renderer.create(<FilmOverview {...mockProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
