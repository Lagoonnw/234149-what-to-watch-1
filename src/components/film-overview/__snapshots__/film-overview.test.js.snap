// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Film overview correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        894 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      Some text
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: John Dow
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: Actor, Actress
      </strong>
    </p>
  </div>,
]
`;
